File: tests/test-net/src/main/java/pize/tests/net/packet/MessagePacket.java
package pize.tests.net.packet;

import pize.net.tcp.packet.IPacket;
import pize.util.io.PizeInputStream;
import pize.util.io.PizeOutputStream;

import java.io.*;

public class MessagePacket extends IPacket{
    
    public static final int PACKET_TYPE_ID = 54;
    
    public MessagePacket(){
        super(PACKET_TYPE_ID);
    }
    
    
    private String message;
    
    public MessagePacket(String message){
        this();
        this.message = message;
    }
    
    public String getMessage(){
        return message;
    }
    
    
    @Override
    protected void write(PizeOutputStream stream) throws IOException{
        stream.writeUTF(message);
    }
    
    @Override
    public void read(PizeInputStream stream) throws IOException{
        message = stream.readUTF();
    }
    
}

File: tests/test-net/src/main/kotlin/pize/tests/net/packet/MessagePacket.kt
package pize.tests.net.packet

import pize.net.tcp.packet.IPacket
import pize.util.io.PizeInputStream
import pize.util.io.PizeOutputStream
import java.io.IOException

class MessagePacket() : IPacket<Any?>(PACKET_TYPE_ID) {
    var message: String? = null
        private set

    constructor(message: String?) : this() {
        this.message = message
    }

    @Throws(IOException::class)
    override fun write(stream: PizeOutputStream?) {
        stream!!.writeUTF(message)
    }

    @Throws(IOException::class)
    override fun read(stream: PizeInputStream?) {
        message = stream!!.readUTF()
    }

    companion object {
        const val PACKET_TYPE_ID = 54
    }
}
